go/carmen/carmen.go

```diff
@@ -277,6 +277,9 @@ type TransactionContext interface {
 	// that gets reset after each transaction.
 	SetTransientState(Address, Key, Value)
 
+	// HasEmptyStorage returns true if account has empty storage.
```

Comment:
The term "empty" has lead to a bit of confusion in the past, in particular since the storage of an account is not really empty. In fact, the storage of an account always has 2^256 elements in it, they may just be all zero.

We could rename this function to "HasZeroState" or "HasAllZeroState", but I guess "HasEmptyState" reads nicer. If we stick to that, however, we should explain in the documentation that an "empty" state is a state where all values are zero.